### PR TITLE
[azure] add ci profiles for worker marketplace image on 4.8-4.10

### PR DESCRIPTION
**Pull request by jinyunma**
1. add azure-ipi-private-worker-marketimage jobs to 4.8/4.9/4.10 amd64 nightly/stable
2. remove azure-ipi-private jobs from 4.8/4.9/4.10 amd64 nightly/stable

### CODE REVIEW

**Comment by jinyunma** [comment]
cc @jianlinliu @shellyyang1989 @MayXuQQ to review

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 1]
```diff
@@ -1080,7 +1080,8 @@ tests:
     cluster_profile: azure-marketplace-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
-      E2E_RUN_TAGS: '@azure-ipi and @network-openshiftsdn and not @fips'
+      E2E_RUN_TAGS: '@azure-ipi and @disconnected and @network-openshiftsdn and not
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1083]: Why this job need to run `disconnected` test case?

**Comment by jianlinliu** [comment]
I guess we also need to add some upgrade jobs for marketplace image testing.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml (hunk 1 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1083]: I saw previous azure-ipi-private job includes this tag, here `@disconnected` means cluster running in a restricted network? if so, might be removed.

**Comment by jinyunma** [comment]
> I guess we also need to add some upgrade jobs for marketplace image testing.

Checked with @shellyyang1989 , @JianLi-RH will help to add upgrade jobs.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1083]: > here @disconnected means cluster running in a restricted network
I think so, I guess we need to remove "@disconnected"

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1083]: @liangxia can you help us confirm that?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1083]: Generally, we only need tag disconnected on disconnected clusters.

Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml]: thanks for the confirmation, removed tag "@disconnected"

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.9__amd64-stable.yaml [hunk 2]
```diff
@@ -341,15 +341,27 @@ tests:
...
   steps:
-    cluster_profile: azure-qe
+    cluster_profile: azure-marketplace-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
     test:
     - ref: cucushift-installer-check-cluster-health
-    workflow: cucushift-installer-rehearse-azure-ipi-private
+    workflow: cucushift-installer-rehearse-azure-ipi-private-workers-marketimage
+- as: azure-ipi-private-workers-marketimage-p3-f28-destructive
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.9__amd64-stable.yaml:353]: Based on latest test strategy for stable payloads, we are making a change to replace e2e with healthcheck, so the normal job and destructive job actually only do install the cluster and cluster health check, which make them the same now. So the destructive job is no longer needed here.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml [hunk 3]
```diff
@@ -732,30 +732,29 @@ tests:
...
-        @fips'
+      E2E_RUN_TAGS: '@azure-ipi and @network-openshiftsdn and not @fips'
       TAG_VERSION: '@4.8'
     test:
     - chain: openshift-e2e-test-qe
-    workflow: cucushift-installer-rehearse-azure-ipi-private
-- as: azure-ipi-private-p3-f360-destructive
-  cron: 14 9 26 3 *
+    workflow: cucushift-installer-rehearse-azure-ipi-private-workers-marketimage
+- as: azure-ipi-private-workers-marketimage-p3-f360-destructive
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml:746]: I guess we can drop the destructive testing now per our latest stable test strategy.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.9__amd64-stable.yaml (hunk 2 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.9__amd64-stable.yaml]: got it, removed.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml (hunk 3 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml:746]: I didn't see the change on 4.8, and testrun `openshift-e2e-test-qe` is still there for other jobs, do I need to drop destructive job here?

Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml]: removed

**Comment by JianLi-RH** [comment]
I am working on the upgrade part, here is the PR: https://github.com/openshift/release/pull/37922
cc @jianlinliu @jinyunma @shellyyang1989 @liangxia

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml (hunk 3 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml]: Yeah, @liangxia is update the existing one in another PR.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml [hunk 4]
```diff
@@ -958,30 +958,29 @@ tests:
...
-  cron: 42 20 14 * *
+    workflow: cucushift-installer-rehearse-azure-ipi-private-workers-marketimage
+- as: azure-ipi-private-workers-marketimage-p3-f28-destructive
+  cron: 57 23 7 * *
   steps:
-    cluster_profile: azure-qe
+    cluster_profile: azure-marketplace-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
       E2E_RUN_TAGS: '@azure-ipi and @disconnected and @network-openshiftsdn and not
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml:978]: I guess we can drop `@disconnected` together.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml (hunk 3 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.8__amd64-stable.yaml]: got it

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml (hunk 4 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml]: thanks for reveiw, missed in some job config, should be dropped now on all related jobs config.

**Pull request merged**